api/v1beta1/cryostat_conversion.go

```diff
@@ -0,0 +1,590 @@
+// Copyright The Cryostat Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	operatorv1beta2 "github.com/cryostatio/cryostat-operator/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+var _ conversion.Convertible = &Cryostat{}
+
+// ConvertTo converts this Cryostat to the Hub version (v1beta2).
+func (src *Cryostat) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*operatorv1beta2.Cryostat)
+
+	// Copy ObjectMeta as-is
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Convert existing Spec fields
+	convertSpecTo(&src.Spec, &dst.Spec)
+
+	// Convert existing Status fields
+	convertStatusTo(&src.Status, &dst.Status)
+
+	// Maintain the previous behaviour by using the CR's namespace as the sole target namespace
+	dst.Spec.TargetNamespaces = []string{src.Namespace}
+	dst.Status.TargetNamespaces = []string{src.Namespace}
+
+	return nil
+}
+
+func convertSpecTo(src *CryostatSpec, dst *operatorv1beta2.CryostatSpec) {
+	dst.Minimal = src.Minimal
+	dst.EnableCertManager = src.EnableCertManager
+	dst.TrustedCertSecrets = convertCertSecretsTo(src.TrustedCertSecrets)
+	dst.EventTemplates = convertEventTemplatesTo(src.EventTemplates)
+	dst.StorageOptions = convertStorageOptionsTo(src.StorageOptions)
+	dst.ServiceOptions = convertServiceOptionsTo(src.ServiceOptions)
+	dst.NetworkOptions = convertNetworkOptionsTo(src.NetworkOptions)
+	dst.ReportOptions = convertReportOptionsTo(src.ReportOptions)
+	dst.MaxWsConnections = src.MaxWsConnections
```

Comment:
I don't think we have an equivalent `MaxWsConnections` config in 3.0 and I'm not sure if we should bother reimplementing this. I think this is an old holdover from when the UI didn't work too well with concurrent users in the 1.0 days.